shared/events.js

```diff
@@ -257,19 +257,41 @@ export class GroupContract extends HashableContract {
     ['openMembership', 'bool'],
     ['memberApprovalPercentage', 'uint32'],
     ['memberRemovalPercentage', 'uint32'],
+    ['incomeProvided', 'float'],
     ['contributionPrivacy', 'string'],
     ['founderUsername', 'string']
   ])
   static vuex = GroupContract.Vuex({
-    state: { votes: [], payments: [] },
+    state: { votes: [], payments: [], members: [], invitees: [] },
     mutations: {
       Payment (state, data) { state.payments.push(data) },
-      Vote (state, data) { state.votes.push(data) }
+      Vote (state, data) { state.votes.push(data) },
+      RecordInvitation (state, data) { state.invitees.push(data.username) },
+      DeclineInvitation (state, data) {
+        let index = state.invitees.findIndex(username => username === data.username)
+        if (index > -1) { state.invitees.splice(index, 1) }
+      },
+      AcceptInvitation (state, data) {
+        let index = state.invitees.findIndex(username => username === data.username)
+        if (index > -1) {
+          state.invitees.splice(index, 1)
+          state.members.push(data.username)
+        }
+      },
+      AcknowledgeFounder (state, data) {
+        if (!state.members.find(username => username === data.username)) { state.members.push(data.username) }
+      }
     }
   })
   constructor (data: JSONObject, parentHash?: string) {
     super({...data, creationDate: new Date().toISOString()}, parentHash)
   }
+  toVuexState () {
+    let state = super.toVuexState()
+    // Place founder in group members before returning the initial state
+    state.members.push(state.founderUsername)
+    return state
+  }
```

Comment:
Hmm. Interesting way of doing this. I think it's fine, can't think of a better way of doing it at the moment...